dev-client/src/screens/SoilScreen/ColorScreen/components/ManualWorkflow.tsx

```diff
@@ -0,0 +1,213 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {SoilPitInputScreenProps} from 'terraso-mobile-client/screens/SoilScreen/components/SoilPitInputScreenScaffold';
+import {
+  Box,
+  Column,
+  Heading,
+  Paragraph,
+  Row,
+} from 'terraso-mobile-client/components/NativeBaseAdapters';
+import {useTranslation} from 'react-i18next';
+import {pitMethodSummary} from 'terraso-mobile-client/screens/SoilScreen/utils/renderValues';
+import {useDispatch, useSelector} from 'terraso-mobile-client/store';
+import {selectDepthDependentData} from 'terraso-client-shared/selectors';
+import {SwitchWorkflowButton} from 'terraso-mobile-client/screens/SoilScreen/ColorScreen/components/SwitchWorkflowButton';
+import {
+  updateDepthDependentSoilData,
+  SoilColorHue,
+  ColorHueSubstep,
+  soilColorHues,
+  ColorValue,
+  ColorChroma,
+} from 'terraso-client-shared/soilId/soilIdSlice';
+import {Icon} from 'terraso-mobile-client/components/Icons';
+import {useCallback, useMemo, useState} from 'react';
+import {Select} from 'terraso-mobile-client/components/inputs/Select';
+import {
+  parseMunsellHue,
+  renderMunsellHue,
+} from 'terraso-mobile-client/screens/SoilScreen/ColorScreen/utils/munsellConversions';
+import Animated, {LinearTransition} from 'react-native-reanimated';
+import {StyleSheet} from 'react-native';
+import {
+  ColorUpdate,
+  PartialColor,
+  updateColorSelections,
+  validProperties,
+} from 'terraso-mobile-client/screens/SoilScreen/ColorScreen/utils/soilColorValidation';
+
+export const ManualWorkflow = (props: SoilPitInputScreenProps) => {
+  const {t} = useTranslation();
+  const dispatch = useDispatch();
+  const data = useSelector(selectDepthDependentData(props));
+  const {complete} = pitMethodSummary(t, data, 'soilColor');
+
+  const {hue: initialHue, substep: initialSubstep} =
+    typeof data?.colorHue === 'number'
+      ? renderMunsellHue({
+          colorHue: data?.colorHue,
+          colorChroma: data?.colorChroma,
+        })
+      : {substep: null, hue: null};
+
+  const [color, setColor] = useState<PartialColor>({
+    hue: initialHue as SoilColorHue | null,
+    substep: initialSubstep,
+    value: (data?.colorValue ?? null) as ColorValue | null,
+    chroma: (data?.colorChroma ?? null) as ColorChroma | null,
+  });
+
+  const updateColor = useCallback(
+    (update: ColorUpdate) => {
+      const newColor = updateColorSelections(color, update);
+      setColor(newColor);
+      dispatch(
+        updateDepthDependentSoilData({
+          siteId: props.siteId,
+          depthInterval: props.depthInterval.depthInterval,
+          colorHue:
+            newColor.hue === 'N'
+              ? 0
+              : newColor.hue && newColor.substep
```

Comment:
comment: let's see if we can clean up this bit too, since it gets pretty dense here